node/connection_base.js

```diff
@@ -230,6 +237,13 @@ TChannelConnectionBase.prototype.buildResponse = function buildResponse(req, opt
             state: req.res.state
         }));
     }
+    options = extend({
```

Comment:
Is there any reason to move to buildOutResponse? Note that both self-connection and connection need it.